File: IT/Eccezioni/Esercizi/Python/esercizio4.py
# file esercizio4.py

from __future__ import print_function

nums = ["1", "2", "", "3", "abc", "7", "def"]
print(nums)

tot = 0

try:
	for n in nums:
		tot += int(n)
except ValueError as e:
	print(e)

print("Somma dei numeri presenti: {}".format(tot))
print("Fine del programma.")

File: IT/Eccezioni/Esercizi/Java/Esercizio4.java
/* file Esercizio4.java */

import java.util.Arrays;

public class Esercizio4 {
	public static void main(String [] args) {
		String [] nums = {"1", "2", "", "3", "abc", "7", "def"};
		System.out.println(Arrays.toString(nums));

		int tot = 0;

		try {
			for (String n : nums) {
				tot += Integer.parseInt(n);
			}
		}
		catch (NumberFormatException e) {
			System.out.println(e.getMessage());
		}

		System.out.println("Somma dei numeri presenti: " + tot);
		System.out.println("Fine del programma.");
	}
}

